programs/escrow/src/lib.rs

```diff
@@ -131,48 +166,76 @@ pub struct EscrowAccount {
 #[instruction(amount: u64, vault_authority_bump: u8)]
 pub struct Deposit<'info> {
     pub mint: Account<'info, Mint>,
-    #[account(mut, signer)]
-    pub giver: AccountInfo<'info>,
+    #[account(mut)]
+    pub giver: Signer<'info>, // I think more common practice to just mark signer like so.
     pub taker: AccountInfo<'info>,
     #[account(
         seeds = [ESCROW_PDA_SEED],
         bump = vault_authority_bump
     )]
     pub vault_authority: AccountInfo<'info>,
-    #[account(mut, constraint = giver_token_account.mint == mint.key() && giver_token_account.amount >= amount)]
+    // I think it is best practice to be as verbose with constraints as possible so this is good.
+    // But it is useful to note that if giver_token_account was not of the Mint account mint, the
+    // instruction would fail at the transfer CPI, rather than here.
+    #[account(
+        mut,
+        constraint = giver_token_account.mint == mint.key() && giver_token_account.amount >= amount,
+        constraint = giver_token_account.owner == giver.key(),
+    )]
     pub giver_token_account: Account<'info, TokenAccount>,
-    #[account(constraint = taker_token_account.mint == mint.key())]
+    #[account(
+        constraint = taker_token_account.mint == mint.key() @ ErrorCode::ExampleError, // This is nice to have custom errors on constraints rather than the generic anchor constraints error.
+        constraint = giver_token_account.owner == taker.key(),
+    )]
+    // We don't need this - I don't think the giver requires information about the taker's token
```

Comment:
agreed, taken it out and having some issues: `Error: failed to send transaction: Transaction simulation failed: Error processing Instruction 0: Cross-program invocation with unauthorized signer or writable account`